File: observer/observer.py
import abc

class Observer(abc.ABC):
    @abc.abstractmethod
    def update(self):
        pass

class InstagramAccount:
    def __init__(self, username):
        self.username = username
        self.__followers = []
    
    def attach(self, observer):
        self.__followers.append(observer)
    
    def detach(self, observer):
        self.__followers.remove(observer)
    
    def notify(self):
        for follower in self.__followers:
            follower.update()
    
    def add_new_post(self, message):
        print(f"New post from {self.username}: {message}")
        self.notify()

class Follower(Observer):
    def __init__(self, username):
        self.username = username

    def update(self):
        print(f"New post on Instagram: {self.username}")

def main():
    arayaroma = InstagramAccount("Daniel")
    follower_one = Follower("Cesar")
    follower_two = Follower("Karla")

    arayaroma.attach(follower_one)
    arayaroma.attach(follower_two)

    arayaroma.add_new_post("I see you there, observer.") 

if __name__ == '__main__':
    main()

File: observer/observer.cpp
#include <iostream>
#include <list>
#include <string>

class Observer {
public:
  virtual void update() = 0;
};

class Follower : public Observer {
private:
  std::string username;

public:
  Follower(std::string username) { this->username = username; }

  void update() override {
    std::cout << "New post on Instagram: " << username << std::endl;
  }
};

class InstagramAccount {
private:
  std::string username;
  std::list<Observer *> followers;

public:
  InstagramAccount(std::string username) { this->username = username; }

  void attach(Observer *observer) { followers.push_back(observer); }

  void detach(Observer *observer) { followers.remove(observer); }

  void notify() {
    for (auto follower : followers) {
      follower->update();
    }
  }

  void addNewPost(std::string message) {
    std::cout << "New post from " << username << ": " << message << std::endl;
    notify();
  }
};

int main() {
  InstagramAccount arayaroma("Daniel");
  Follower followerOne("Cesar");
  Follower followerTwo("Karla");

  arayaroma.attach(&followerOne);
  arayaroma.attach(&followerTwo);

  arayaroma.addNewPost("I see you there, in C++.");

  return EXIT_SUCCESS;
}